### ISSUE TITLE
Wine is only 64 bit on x86_64, can't use 32 bit prefix neither WINEARCH=win32

### ISSUE THREAD

**Issue by rafi16jan** [skycocker/chromebrew#3049]
## Before you submit an issue
Please look up our issues, maybe your problem was mentioned before.

## Description
Wine is only 64 bit on x86_64, can't use 32 bit prefix neither WINEARCH=win32
See #3048 (for reference)

If you executed a specific command:
```shell
WINEPREFIX=.wine32 WINEARCH=win32 wine 32bitapp.exe
```

or even

```shell
WINEPREFIX=.wine32 WINEARCH=win32 winecfg
```

Output:
```shell
.wine32 is a 32-bit installation, it cannot support 64-bit applications.
```

## Versions and system information
chromebrew: 1.0.4

ChromeOS: 
```

CHROMEOS_RELEASE_APPID={49BA18F3-93DB-4F43-B966-3BBC57881C42}
CHROMEOS_BOARD_APPID={49BA18F3-93DB-4F43-B966-3BBC57881C42}
CHROMEOS_CANARY_APPID={90F229CE-83E2-4FAF-8479-E368A34938B1}
DEVICETYPE=CHROMEBOOK
CHROMEOS_ARC_VERSION=5029284
CHROMEOS_ARC_ANDROID_SDK_VERSION=25
GOOGLE_RELEASE=11021.42.2.1
CHROMEOS_DEVSERVER=https://devserver.fydeos.com:9999
CHROMEOS_RELEASE_BOARD=amd64-fydeos
CHROMEOS_RELEASE_BUILD_NUMBER=11021
CHROMEOS_RELEASE_BRANCH_NUMBER=42
CHROMEOS_RELEASE_CHROME_MILESTONE=70
CHROMEOS_RELEASE_PATCH_NUMBER=2
CHROMEOS_RELEASE_TRACK=stable-channel
CHROMEOS_RELEASE_DESCRIPTION=11021.42.2.1 (Release Build v5.1) stable-channel amd64-fydeos
CHROMEOS_RELEASE_BUILD_TYPE=Release Build
CHROMEOS_RELEASE_NAME=Chromium OS
CHROMEOS_RELEASE_VERSION=11021.42.2.1
CHROMEOS_AUSERVER=https://up.fydeos.com/service/update2
```
Model: Not a chromebook (Fyde OS)

Architecture: `x86_64`

## Addtional information
-wine executable is a symlink to wine64, should be a separate wine 32 bit binary
-tried the build_cros_gcc script without success (it doesn't do anything, but changing the if else about ARCH, JOBS, PREFIX make it run, but the make command gives a C error)

What I think needs to be done:
- Either make a prebuilt binaries or
- Provide a way to cross compile to i686

---

## That's it
Thank you for submitting your issue.
When done, please delete the parts of this template which you don't need or these, which are only for guidance.

**Comment by JL2210** [comment]
Anyway, the `build_cross_gcc` script was mainly meant for bootstrapping a Chromebrew installation, not general cross compilation.

**Comment by rafi16jan** [comment]
@JL2210 I think bootstrapping crew or even chroot (and install separate crew with ARCH changed to i686) is needed here. I remember I use to setup i686 crouton instead of amd64, and it works. It should work with chromebrew too with extra steps

**Comment by rafi16jan** [comment]
Reading from the FAQ, you guys have a dpkg package? Can we install .deb wine package or i686 gcc?

**Comment by JL2210** [comment]
Most certainly not. First of all, we're going to remove support for installing packages with `dpkg`. Second of all, if we were able to install packages with `dpkg`, those packages would fail as they install files in `/usr` and `/`. Third of all, even if we did install those packages, they most certainly would not work because of the missing 32-bit dynamic linker and system C libraries.

**Comment by JL2210** [comment]
@rafi16jan I'm working on a fix for this, but for now just run [ReactOS](https://reactos.org) in QEMU (`crew install qemu`).

**Comment by rafi16jan** [comment]
@JL2210 I think I won't use qemu, my macbook air fan turn into a jet engine if I play an Android game. I'll just wait.

Anyway, is this relevant? [](https://wiki.winehq.org/Building_Wine#Shared_WoW64)

**Comment by JL2210** [comment]
Yes, however it won't work. Trust me, I've tried and tried. Your best bet is to install another operating system if all you want to do is run Windows apps. Anyway, ReactOS is a very light operating system, so I really suggest you give it a shot in QEMU.

**Comment by rafi16jan** [comment]
@JL2210 Well I checked and ReactOS have poor directx support. So maybe I'll just emulate Windows XP instead.

**Comment by JL2210** [comment]
@rafi16jan To install 32-bit Wine on your Chromebook:
Download this script: 
```
cd ~/Downloads && \
wget https://gist.github.com/JL2210/f51a3d5a7a57900983b706949eba9f74/raw/b87fa07e0cb8aa80da3e2ea233f9841c9be1eb61/bootstrap.sh
```
Run it: 
```
bash bootstrap.sh
```
When asked if you want to install a 32-bit Prefix, type `y`.
Grab a coffee. This will take a while.
After it's done, run this command:
```
/usr/local/prefix32/startprefix
```
After doing so, run this command:
```
emerge app-emulation/wine
```
Then, run `wine <application>` and be happy.

**Comment by TheElf01** [comment]
> Grab a coffee. This will take a while.


Hi, how much is a "while"?  or how much coffe i need?

my chromebook is compiling since 2 days ago, and still did not finish ,  any idea? 

thanks

**Comment by JL2210** [comment]
That's pretty normal. Leave it going overnight if possible, and try not to do much else (but it's fine if you don't leave it overnight or do something else; it'll just be slower). I tried to make it more minimal, but these are the bare essentials.

**Comment by TheElf01** [comment]
> That's pretty normal. Leave it going overnight if possible, and try not to do much else (but it's fine if you don't leave it overnight or do something else; it'll just be slower). I tried to make it more minimal, but these are the bare essentials.

No problem, perfect, just wanna know 

My only concern now is storage space,  Before start I erase everything, and left 7.2GB free, and now i have 1.1GB....  hope will  be enough... if not will be sad a fail after 2 days compiling because out of space jeje

thanks

**Comment by TheElf01** [comment]
> That's pretty normal. Leave it going overnight if possible, and try not to do much else (but it's fine if you don't leave it overnight or do something else; it'll just be slower). I tried to make it more minimal, but these are the bare essentials.

Sorry to double post...

After more than 2 days,  my Chromebook just sudenly power off... battery was 100%, it was AC connected.. just i found powered off

Is normal?

I write /usr/local/prefix32/startprefix like you say next step

But no prefix32 directory in usr/local

**Comment by JL2210** [comment]
Um... No. After it fails, it fails. Do you have any big files or packages you can remove?

**Comment by TheElf01** [comment]
> Um... No. After it fails, it fails. Do you have any big files or packages you can remove? (Run `rm -rf /usr/local/gentoo{,32}` to free up some space)

thanks... well start from begining... another 2 days

I dont think is about free space, when the machine power off,  still the HDD has 900MB free

**Comment by eudy97** [comment]
> @rafi16jan To install 32-bit Wine on your Chromebook: Download this script:
> 
> ```
> cd ~/Downloads && \
> wget https://gist.github.com/JL2210/f51a3d5a7a57900983b706949eba9f74/raw/b87fa07e0cb8aa80da3e2ea233f9841c9be1eb61/bootstrap.sh
> ```
> 
> Run it:
> 
> ```
> bash bootstrap.sh
> ```
> 
> When asked if you want to install a 32-bit Prefix, type `y`. Grab a coffee. This will take a while. After it's done, run this command:
> 
> ```
> /usr/local/prefix32/startprefix
> ```
> 
> After doing so, run this command:
> 
> ```
> emerge app-emulation/wine
> ```
> 
> Then, run `wine <application>` and be happy.

Does this still work? I get a "could not resolve host error".

**Comment by saltedcoffii** [comment]
bump. This requires a multilib glibc, which may be a nightbare.

**Comment by satmandu** [comment]
There's zero chance I'm going to work on modifying the glibc install to support multilib. It would be better to setup a chrooted multilib glibc to handle this... so better to use crouton for that, or crostini, since that can handle a multilib glibc via crosvm.

**Comment by rafi16jan** [comment]
> Wine's [WoW64 support](https://gitlab.winehq.org/wine/wine/-/releases/wine-9.0#wow64) is shaping up nicely, particularly in 9.0, so perhaps that's a viable option.

This is very promising. Maybe I will try to make a fork of the wine.rb
